File: apps/desktop/frontend/src/utils/tray.ts
import { invoke } from '@tauri-apps/api/core';

/**
 * 隐藏窗口到系统托盘
 */
export async function hideToTray(): Promise<void> {
  try {
    await invoke('hide_to_tray');
  } catch (error) {
    console.error('Failed to hide to tray:', error);
    throw error;
  }
}

/**
 * 从系统托盘显示窗口
 */
export async function showFromTray(): Promise<void> {
  try {
    await invoke('show_from_tray');
  } catch (error) {
    console.error('Failed to show from tray:', error);
    throw error;
  }
}

/**
 * 检查窗口是否可见
 */
export async function isWindowVisible(): Promise<boolean> {
  try {
    return await invoke('is_window_visible');
  } catch (error) {
    console.error('Failed to check window visibility:', error);
    return false;
  }
}

/**
 * 退出应用
 */
export async function quitApp(): Promise<void> {
  try {
    await invoke('quit_app');
  } catch (error) {
    console.error('Failed to quit app:', error);
    throw error;
  }
}

File: apps/desktop/src-tauri/src/commands/tray.rs
//! 系统托盘相关命令

use tauri::{AppHandle, Manager, Runtime};

/// 隐藏窗口到托盘
#[tauri::command]
pub fn hide_to_tray<R: Runtime>(app: AppHandle<R>) -> Result<(), String> {
    if let Some(window) = app.get_webview_window("main") {
        window.hide().map_err(|e| e.to_string())?;
    }
    Ok(())
}

/// 显示窗口
#[tauri::command]
pub fn show_from_tray<R: Runtime>(app: AppHandle<R>) -> Result<(), String> {
    if let Some(window) = app.get_webview_window("main") {
        window.show().map_err(|e| e.to_string())?;
        window.set_focus().map_err(|e| e.to_string())?;
    }
    Ok(())
}

/// 检查窗口是否可见
#[tauri::command]
pub fn is_window_visible<R: Runtime>(app: AppHandle<R>) -> Result<bool, String> {
    if let Some(window) = app.get_webview_window("main") {
        window.is_visible().map_err(|e| e.to_string())
    } else {
        Ok(false)
    }
}

/// 退出应用
#[tauri::command]
pub fn quit_app<R: Runtime>(app: AppHandle<R>) {
    app.exit(0);
}
